backend/tests/integration-async/records/indexer/test_authors.py

```diff
@@ -125,3 +125,75 @@ def test_record_created_through_api_is_indexed(
         },
     ]
     retry_until_matched(steps)
+
+
+def test_indexer_updates_authors_papers_when_name_changes(
+    app, celery_app_with_context, celery_session_worker, retry_until_matched
+):
+    author_data = faker.record("aut")
+    author = AuthorsRecord.create(author_data)
+    db.session.commit()
+    es.indices.refresh("records-authors")
+    author_cn = author["control_number"]
+
+    lit_data = {
+        "authors": [
+            {
+                "record": {
+                    "$ref": f"https://labs.inspirehep.net/api/authors/{author_cn}"
+                },
+                "full_name": author["name"]["value"],
+            }
+        ]
+    }
+    lit_data = faker.record("lit", data=lit_data)
+
+    lit_1 = LiteratureRecord.create(lit_data)
+    db.session.commit()
+    steps = [
+        {"step": es.indices.refresh, "args": ["records-authors"]},
+        {
+            "step": es.search,
+            "args": ["records-hep"],
+            "expected_result": {
+                "expected_key": "hits.total.value",
+                "expected_result": 1,
+            },
+        },
+        {
+            "step": es.search,
+            "args": ["records-hep"],
+            "expected_result": {
+                "expected_key": "hits.hits[0]._source.facet_author_name[0]",
+                "expected_result": f"{author['control_number']}_{author['name']['value']}",
+            },
+        },
+    ]
+    retry_until_matched(steps)
```

Comment:
I'd try to avoid using the ``retry_until_matched`` tbh, can we just timeout instead? I know in the past we were avoiding it but still is better ☺️

Reply:
Can I ask why?

Reply:
because the logic of ``retry_until_matched`` it's wrong, if it doesn't match it goes through **all** the steps again and again. That's why we marked it as deprecated.

Reply:
ok